tests/unit/changeset-test.js

```diff
@@ -830,4 +838,166 @@ test('content can be an empty hash', function(assert) {
   let dummyChangeset = new Changeset(emptyObject, dummyValidator);
 
   assert.equal(dummyChangeset.toString(), 'changeset:[object Object]');
-})
+});
+
+test('it works with nested keys', function(assert) {
+  let expectedResult = {
+    org: {
+      asia: { sg: 'sg' },
+      usa: {
+        ca: 'ca',
+        ny: 'ny',
+        ma: { name: 'Massachusetts' }
+      }
+    }
+  };
+  set(dummyModel, 'org', {
+    asia: { sg: null },
+    usa: {
+      ca: null,
+      ny: null,
+      ma: { name: null }
+    }
+  });
+  let dummyChangeset = new Changeset(dummyModel, dummyValidator);
+  dummyChangeset.set('org.asia.sg', 'sg');
+  dummyChangeset.set('org.usa.ca', 'ca');
+  dummyChangeset.set('org.usa.ny', 'ny');
+  dummyChangeset.set('org.usa.ma', { name: 'Massachusetts' });
+  dummyChangeset.execute();
+  assert.deepEqual(get(dummyChangeset, 'change'), expectedResult, 'should have correct shape');
+  assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
```

Comment:
These assertions seem redundant since we're already doing a `.deepEqual` above. Let's remove these?

Reply:
removed the tests below, and pulled out the rollback tests and have those just in their own now.